keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -935,6 +961,11 @@ public BooleanTensor greaterThanOrEqual(double value) {
         return fromMask(tensor.gte(value), copyOf(getShape(), getShape().length));
     }
 
+    @Override
+    public BooleanTensor isNaN() {
+        return TensorValidator.thatChecksForNaN().check(this).not();
```

Comment:
I find reading this a bit strange. A method called `isNaN` and then we're calling a method called `thatChecksForNaN` and we're inverting it. Can we change this to `TensorValidator. thatExpectsNotToFindNan().check(this).not()`

Reply:
This is obsolete now as I've changed (improved) how this method works. It now does some direct tensor stuff.